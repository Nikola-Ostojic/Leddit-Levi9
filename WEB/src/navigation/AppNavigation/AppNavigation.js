import React from "react";
import { BrowserRouter } from "react-router-dom";

import axiosInstance, { cancelTokenSource } from "utils/ApiCall";
import Navbar from "navigation/Navbar";
import Drawer from "navigation/Drawer";
import AppRoutes from "navigation/AppRoutes";

class AppNavigation extends React.Component {
  state = {
    drawerOpen: false
  };

  componentDidMount() {
    const { checkUser, getToken } = this.props;
    checkUser();
    axiosInstance.interceptors.request.use(async (config) => {
      if (config.requiresAuth) {
        config.cancelToken = cancelTokenSource.token;
        const token = await getToken(cancelTokenSource.cancel);
        config.headers.Authorization = `Bearer ${token}`;
      }
      return config;
    });
  }

  render() {
    const { drawerOpen } = this.state;
    return (
      <BrowserRouter>
        <Navbar
          drawerOpen={drawerOpen}
          setDrawerOpen={(open) => this.setState({ drawerOpen: open })}
        />
        <Drawer
          drawerOpen={drawerOpen}
          setDrawerOpen={(open) => this.setState({ drawerOpen: open })}
        />
        <AppRoutes />
      </BrowserRouter>
    );
  }
}
export default AppNavigation;
