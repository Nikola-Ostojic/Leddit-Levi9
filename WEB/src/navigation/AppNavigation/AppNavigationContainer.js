import { connect } from "react-redux";

import { getToken, checkUser } from "modules/Auth/AuthActions";

import AppNavigation from "./AppNavigation";

export default connect(null, { getToken, checkUser })(AppNavigation);
