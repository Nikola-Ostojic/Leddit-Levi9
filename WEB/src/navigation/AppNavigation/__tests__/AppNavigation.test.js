import React from "react";
import { shallow } from "enzyme";

import AppNavigation from "../AppNavigation";

describe("AppNavigation tests", () => {
  const mockedProps = {
    checkUser: jest.fn()
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<AppNavigation {...mockedProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call check user after mount", () => {
    shallow(<AppNavigation {...mockedProps} />);
    expect(mockedProps.checkUser).toHaveBeenCalled();
  });
});
