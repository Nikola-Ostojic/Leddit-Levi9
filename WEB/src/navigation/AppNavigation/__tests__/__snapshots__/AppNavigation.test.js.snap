// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation tests Should match snapshot 1`] = `
<BrowserRouter>
  <Connect(withRouter(Navbar))
    drawerOpen={false}
    setDrawerOpen={[Function]}
  />
  <withRouter(Drawer)
    drawerOpen={false}
    setDrawerOpen={[Function]}
  />
  <AppRoutes />
</BrowserRouter>
`;
